restorers/model/nafnet/nafnet.py

```diff
@@ -0,0 +1,178 @@
+from typing import Optional, Tuple
+
+import tensorflow as tf
+from tensorflow import keras
+from .nafblock import NAFBlock
+
+
+class PixelShuffle(keras.layers.Layer):
+    """
+    PixelShuffle Layer
+
+    Wrapper Class for tf.nn.depth_to_space
+    """
+
+    def __init__(self, upscale_factor: int, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.upscale_factor = upscale_factor
+
+    def call(self, inputs: tf.Tensor, *args, **kwargs) -> tf.Tensor:
+        return tf.nn.depth_to_space(inputs, self.upscale_factor)
+
+
```

Comment:
Could you add a `get_config` fn to all custom models as well ? It helps with distributed training in some instances. [Reference Implementation](https://github.com/SauravMaheshkar/Knowledge-Distillation/blob/ef5b92bd53852df5d920746ba19d870360467e5b/src/backend/tensorflow/modules/affine.py#L46)